### PR TITLE
Add handling parens for extends clause in generator

### LINKED ISSUE #6895: Breaks conditional extends
> Original issue submitted by @johanjisu in https://github.com/babel/minify/issues/739

minify breaks code with conditional extends, such as this:

```js
export default Target => class Events extends (Target || Base) {

}
```

Minification removes () around Target || Base, which breaks it.

(the code above is used for mixins, Base is an existing class)

### PR DESCRIPTION
**Pull request by existentialism**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #6895
| Patch: Bug Fix?          | y
| Major: Breaking Change?  | n
| Minor: New Feature?      | n
| Tests Added + Pass?      | y/y
| Documentation PR         | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: packages/babel-generator/test/fixtures/parentheses/class-extends/expected.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+class A extends class {} {}
+
+class A extends (B ? C : D) {}
+
+class A extends (new B()) {}
+
+class A extends (B, C) {}
+
+class A extends ({}) {}
```
Comment (nicolo-ribaudo) [inline packages/babel-generator/test/fixtures/parentheses/class-extends/expected.js]: Are the parentheses actually needed here?

Comment (existentialism) [reply packages/babel-generator/test/fixtures/parentheses/class-extends/expected.js]: updated!

**Comment by nicolo-ribaudo** [comment]
I didn't mean to approve yet :sweat_smile:

**Comment by johanjisu** [comment]
Thanks guys, that was quickly fixed

**Pull request merged**